src/dataset.ts

```diff
@@ -205,6 +206,7 @@ class Dataset extends ServiceObject {
       baseUrl: '/datasets',
       id: id,
       methods: methods,
+      requestModule: request,
```

Comment:
We should make the common module inherit the parent's `requestModule`. @fhinkel wdyt?

Reply:
Makes sense to me!

Reply:
Instead of specifying it in the constructor? Not sure I see a huge advantage, as you need to specify it either way.

Reply:
Sorry, my terminology was a bit off. `Dataset` is a `ServiceObject`, whose parent is `BigQuery`, a `Service`. Since we set `requestModule` through BigQuery, it would be easier if the ServiceObject children would inherit it to save us from the multiple imports and redundancy.

Reply:
Opened an issue on common: https://github.com/googleapis/nodejs-common/issues/222